treeherder/webapp/api/performance_data.py

```diff
@@ -267,24 +274,78 @@ class AlertSummaryPagination(pagination.PageNumberPagination):
     page_size = 10
 
 
+class PerformanceAlertSummaryFilter(django_filters.FilterSet):
+    id = django_filters.NumberFilter(field_name='id')
+    status = django_filters.NumberFilter(field_name='status')
+    framework = django_filters.NumberFilter(field_name='framework')
+    repository = django_filters.NumberFilter(field_name='repository')
+    alerts__series_signature = django_filters.NumberFilter(field_name='alerts__series_signature')
+    filter_text = django_filters.CharFilter(method='_filter_text')
+    hide_improvements = django_filters.BooleanFilter(method='_hide_improvements')
+    hide_related_and_invalid = django_filters.BooleanFilter(method='_hide_related_and_invalid')
+    with_assignee = django_filters.CharFilter(method='_with_assignee')
+
+    def _filter_text(self, queryset, name, value):
+        sep = Value(' ')
+        words = value.split(' ')
+
+        contains_all_words = [Q(full_name__contains=word) for word in words]
+
+        # Django's distinct(*fields) isn't supported for MySQL
+        # https://code.djangoproject.com/ticket/17974
+        filtered_summaries = (queryset.annotate(
+            full_name=Concat(
+                'alerts__series_signature__suite', sep,
+                'alerts__series_signature__test', sep,
+                'alerts__series_signature__platform__platform', sep,
+                'alerts__series_signature__extra_options', sep,
+                'bug_number', sep,
+                'push__revision',
+                output_field=CharField()))
+                                      .filter(*contains_all_words)
+                                      .values('id')
+                                      .distinct())
+
+        return queryset.filter(id__in=Subquery(filtered_summaries))
```

Comment:
Are these results paginated?

Reply:
Yes, they are automatically paginated by our custom Django-based code. More precisely, [here](https://github.com/mozilla/treeherder/blob/6e09f58bf48b8be2ffdfaa10462a4480ed7069e8/treeherder/webapp/api/performance_data.py#L289).

Reply:
I can't remember if I've mentioned this before, but you might find the [CustomPaginator](https://github.com/mozilla/treeherder/blob/6e09f58bf48b8be2ffdfaa10462a4480ed7069e8/treeherder/webapp/api/pagination.py#L13) useful if you end up profiling this API for performance and see issues. In the past, the count method was performing a full table scan when returning paginated results.